lmdeploy/pytorch/engine/logits_process.py

```diff
@@ -1,12 +1,206 @@
 # Copyright (c) OpenMMLab. All rights reserved.
-from typing import Dict
+from dataclasses import asdict, dataclass
+from typing import Dict, List
 
 import torch
 from transformers.generation.logits_process import LogitsWarper
 
 from ..messages import SamplingParam
 
 
+def _process_temperature(scores: torch.Tensor,
+                         temperature: torch.Tensor,
+                         inplace: bool = True):
+    """process temperature."""
+    if not inplace:
+        scores = scores / temperature[:, None]
```

Comment:
What if `temperature` is 0?

Reply:
I think it's better to check it  at the engine's API, neither pipeline API, nor RESTful API.
Because engine is responsible for sampling, and should validate it before using it.